packages/sdk/src/utils/spinner.ts

```diff
@@ -0,0 +1,59 @@
+import type { Options as OraOptions } from 'ora'
+import ora from 'ora'
+
+const defaultOraOptions: OraOptions = {
+  spinner: 'dots',
+  color: 'cyan',
+  indent: 0,
+  stream: process.stdout,
+}
+
+/**
+ * Methods available to a spinner instance that has already started.
+ */
+export interface SpinnerStarted {
+  success(text?: string): void
+  failure(text?: string): void
+}
+
+/**
+ * Closure that starts a spinner if `enableOutput` is true, and returns a `SpinnerStarted` instance.
+ * Note: the spinner is automatically disabled when `process.env.CI` is defined.
```

Comment:
```suggestion
 * Note: the spinner will only be enabled if the stream is being run inside a TTY context (not spawned or piped) and/or not in a CI environment.
```
From ora README, might be more accurate